### PR TITLE
Added ScanServer tests for different tablet hosting goals

### PR DESCRIPTION
**Pull request by dlmarion**

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/ScanServerIT.java [hunk 1]
```diff
@@ -224,6 +234,110 @@ public void testBatchScannerTimeout() throws Exception {
...
+      }
+
+      try (BatchScanner scanner = client.createBatchScanner(tableName, Authorizations.EMPTY)) {
+        scanner.setRanges(Collections.singleton(new Range()));
+        scanner.setConsistencyLevel(ConsistencyLevel.EVENTUAL);
+        assertEquals(ingestedEntryCount, Iterables.size(scanner),
+            "The scan server scanner should have seen all ingested and flushed entries");
+        // Throws an exception because of the tablets with the NEVER hosting goal
+        scanner.setConsistencyLevel(ConsistencyLevel.IMMEDIATE);
+        assertThrows(RuntimeException.class, () -> Iterables.size(scanner));
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ScanServerIT.java:336]: Could also try scanning the ranges of the table that should work.

```suggestion
        assertThrows(RuntimeException.class, () -> Iterables.size(scanner));
        
        scanner.setRange(new Range(null,"row_0000000003"));
        assertEquals(???, Iterables.size(scanner));
        
         scanner.setRange(new Range("row_0000000008",null));
         assertEquals(???, Iterables.size(scanner));
```

File: test/src/main/java/org/apache/accumulo/test/ScanServerIT.java [hunk 2]
```diff
@@ -224,6 +234,110 @@ public void testBatchScannerTimeout() throws Exception {
...
+
+      try (Scanner scanner = client.createScanner(tableName, Authorizations.EMPTY)) {
+        scanner.setRange(new Range());
+        scanner.setConsistencyLevel(ConsistencyLevel.EVENTUAL);
+        assertEquals(ingestedEntryCount, Iterables.size(scanner),
+            "The scan server scanner should have seen all ingested and flushed entries");
+        // Throws an exception because of the tablets with the NEVER hosting goal
+        scanner.setConsistencyLevel(ConsistencyLevel.IMMEDIATE);
+        assertThrows(RuntimeException.class, () -> Iterables.size(scanner));
+      } // when the scanner is closed, all open sessions should be closed
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ScanServerIT.java:285]: can also add immediates scans of the ranges that should work here

File: test/src/main/java/org/apache/accumulo/test/ScanServerIT.java (hunk 1 above)
Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/ScanServerIT.java:336]: The above comment was meant for the scanner test, but I made it in the batch scanner test.  So the set range call is not correct.

Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/ScanServerIT.java:326]: Added in cabcf0b

File: test/src/main/java/org/apache/accumulo/test/ScanServerIT.java [hunk 3]
```diff
@@ -205,6 +217,124 @@ public void testBatchScannerTimeout() throws Exception {
...
+        // Throws an exception because of the tablets with the NEVER hosting goal
+        scanner.setConsistencyLevel(ConsistencyLevel.IMMEDIATE);
+        assertThrows(RuntimeException.class, () -> Iterables.size(scanner));
+
+        // Test that hosted ranges work
+        Collection<Range> ranges = new ArrayList<>();
+        ranges.add(new Range(null, "row_0000000003"));
+        ranges.add(new Range("row_0000000008", null));
+        scanner.setRanges(ranges);
+        assertEquals(60, Iterables.size(scanner));
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ScanServerIT.java:270]: This is neat, a single batch scan against ranges with diff hosting goals.

**Pull request merged**